src/ff/prime_field.rs

```diff
@@ -171,6 +160,9 @@ macro_rules! field_impl {
                 }
             }
         }
+
+        // Make sure FieldType has a member for this field implementation.
+        const _FIELD_TYPE_VALUE: FieldType = crate::ff::FieldType::$field;
```

Comment:
```suggestion
        const _FIELD_TYPE_VALUE: FieldType = FieldType::$field;
```
?

I like this tho. catch the issue at compile time